Closes #2897 ADAB Vignette

Kcdahn opened the pull request:
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/cran-release/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF and a PNG version of it to the same folder. (The PNG version can be created by taking a screenshot of the PDF version.)
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/cran-release/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

Fanny-Gautier reviewed (changes requested):
Same comments as in the template to be implemented here.

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+
+At this step, we first adjust for whether our source data is IS using SDTM v2.0 standard
+or pre SDTM v1.8.  Version 2.0 replaced ISTESTCD and ISTEST with general ADA categories,
```
```suggestion
or pre SDTM v1.8. Version 2.0 replaced ISTESTCD and ISTEST with general ADA categories,
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+an model or implementation guide for ADA (ADAMB). There is a committee and the
+elements of this template reflect methods from a couple members and can serve
+as a baseline for when a guide is released.  CDISC ADaM conventions and similar
+domains (such as ADPC) are followed.
```
```suggestion
domains (such as `ADPC`) are followed.
```

Kcdahn replied:
OK.

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+One of the important aspects of the dataset is the separate derivation of
+original 'by visit' data, by visit interpreted results and parameter variables
+(like ADEX summary parameters).  A good analogy would be ADEX that has a mix
```
```suggestion
(like `ADEX` summary parameters). A good analogy would be `ADEX` that has a mix
```

Fanny-Gautier commented on vignettes/adab.Rmd:
````diff
@@ -0,0 +1,1583 @@
+```
+
+
+At this step, we first adjust for whether our source data is IS using SDTM v2.0 standard
````
```suggestion
At this step, we first adjust for whether our source data is `IS` using SDTM v2.0 standard
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+in the event IS from pre version 2.0 are used. IS.ISTESTCD, and IS.ISBDAGNT from version 2.0
+could be substituted if desired.
+
+DRUG (similar to ADPC.DRUG) is an assigned variable so IS Analytes can be merged matching 
```
```suggestion
`DRUG` (similar to `ADPC.DRUG`) is an assigned variable so `IS Analytes` can be merged matching 
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+could be substituted if desired.
+
+DRUG (similar to ADPC.DRUG) is an assigned variable so IS Analytes can be merged matching 
+EX.EXTRT value(s) for purpose of computing reference dose and time from first dose.
```
```suggestion
`EX.EXTRT` value(s) for purpose of computing reference dose and time from first dose.
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+based on the ADSL.TRTSDT while ADAB.AFRLT is based on the matching EX.EXTRT for the corresponding 
+IS analyte.
+
+If IS.ISDTC is missing the time part, 00:00 is used for the impute.  Individual companies and 
```
```suggestion
If `IS.ISDTC` is missing the time part, "00:00" is used for the imputation. Individual companies and 
```

Kcdahn commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+Final Key Variables this section:
+Subject By Visit analysis variables (sponsor or molecule specific or optional)
+TFLAGV - By visit Treatment Emergent Category (see *T-Flags).
+PBFLAGV - Post Baseline Status y Visit
```
Corrected for next update.

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+studies may have custom impute methods.  If this is the case, the` derive_vars_dtm()`
+would be either replaced or post-processed to get the desired imputed time part. For example,
+a common method is to use EX.EXSTDTC time part (match merged by IS and EX date parts) 
+then IS time is adjusted for pre-dose sampling planned time.  User would add custom code for this.
```
```suggestion
then the `IS` time is adjusted for pre-dose sampling planned time. User would add custom code for this.
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+At this step, we stage 'EX' data. Filter the data as desired to match 
+the 'IS' analytes. As with 'IS', DRUG is assigned as a common merge var.
+As with 'IS', NFRLT is assigned using VISITDY (or the user could substitute
```
```suggestion
As with 'IS' dataset, `NFRLT` is assigned using `VISITDY` (or the user could substitute
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+At this step, we stage 'EX' data. Filter the data as desired to match 
+the 'IS' analytes. As with 'IS', DRUG is assigned as a common merge var.
+As with 'IS', NFRLT is assigned using VISITDY (or the user could substitute
+EX.VISIT values).
```
```suggestion
`EX.VISIT` values).
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+    EXDOSE >= 0
+  ) %>%
+  mutate(
+    # DRUG is a merge var to map and merge ADA data with EX.EXTRT
```
```suggestion
    # DRUG is a merge variable to map and merge ADA data with EX.EXTRT
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+## Assign Periods, Phases and BASETYPE {#basetype}
+
+At this step compute or assign applicable APERIOD, APHASE and/or BASETYPE.
```
```suggestion
At this step compute or assign applicable `APERIOD`, `APHASE` and/or `BASETYPE` variables.
```

Kcdahn commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+      TRUE ~ NA_integer_
+    )
+  ) %>%
+  arrange(DRUG, BASETYPE, ADATYPE, ADAPARM, !!!get_admiral_option("subject_keys"), )
```
Agreed. Corrected for next update.

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+Note: In ADAB, BASETYPE is Phase or Period but the method for assigning BASE
+could be appended to BASETYPE (i.e. "DOUBLE_BLINDED LAST"). 
+BASETYPE is carried though through the remaining steps of this template as a by 
+variable for computing parameters.  For example, if there is main
```
```suggestion
variable for computing parameters. For example, if there is a main
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+At this step, we assign MRT (Minimum Reportable Titer) and an Optional
+DTL (both are user/protocol specific assigned values).  RESULTC is the
+the standard interpreted result (POSITIVE or NEGATIVE) based valued found
```
```suggestion
the standard interpreted result (`POSITIVE` or `NEGATIVE`) based values found
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+to aid with user review of the progression of the program.
+
+When IS.ISSTERC is "POSITIVE",  IS.ISSTRESN is commonly populated with values >0. 
+However, If ISSTRESC is "POSITIVE" with a missing IS.ISSTRESN, this template provides
```
```suggestion
However, if `ISSTRESC` is "POSITIVE" with a missing `IS.ISSTRESN`, this template provides
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+Final Key Variables this section:
+Subject By Visit analysis variables (sponsor or molecule specific or optional)
+TFLAGV - By visit Treatment Emergent Category (see *T-Flags).
+PBFLAGV - Post Baseline Status y Visit
```
```suggestion
PBFLAGV - Post-Baseline Status y Visit
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+  )
+
+# Use an outer Join to combine baseline with most post results create utility dataset with flag data
+flagdata_init <- full_join(base_data, most_post, by = c(
```
base_data dataset is used only from here while its use is described at line 527. Is it necessary at line 527 or could it be explained further more?

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+The target variable for Nab is NABSTAT (subject Nab Status for the the matching parent analyte)
+In the example template, to be NABSTAT = "Positive" or "Negative", the subject has to 
+ADATAT="Positive" on the main parent matching analyte. If not ADASTAT  = Positive,  NABSTAT is 
```
```suggestion
`ADATAT="Positive"` on the main parent matching analyte. If not `ADASTAT  = "Positive"`, `NABSTAT` is 
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+  )
+
+# Compute NAB Stat using both methods
+# Note: Options 2 added as a placekeeper starter method, adjust as needed for
```
```suggestion
# Note: Option 2 added as a placekeeper starter method, adjust as needed for
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+In this section, we bring together all the prior section variables into a single horizontal dataset
+then compute remaining analysis variables into a  final horizontal dataset 
+by STUDYID, USUBJID, BASETYPE, ADATYPE (Nab or ADA), ADAPARM (Drug specific value for the Nab or ADA test)
+The subject level variable values are repeated on each VISIT/ISDTC by BASETYPE, ADATYPE, ADAPARM.
```
```suggestion
The subject level variable values are repeated on each `VISIT`/`ISDTC` by `BASETYPE`, `ADATYPE`, `ADAPARM`.
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+by STUDYID, USUBJID, BASETYPE, ADATYPE (Nab or ADA), ADAPARM (Drug specific value for the Nab or ADA test)
+The subject level variable values are repeated on each VISIT/ISDTC by BASETYPE, ADATYPE, ADAPARM.
+
+The remaining pertinent analysis variables  computed in this section are:
```
```suggestion
The remaining pertinent analysis variables computed in this section are:
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+ENHANCED - Positive at Baseline to Positive Post-Baseline where CHG >= 0.60
+TRANADA  - Transient ADA  (either of): 
+             INDUCED and Positive at any Post Baseline other than Last
+             INDUCED and multiple post baseline positive (other than last)
```
```suggestion
             `INDUCED` and multiple post-baseline positive (other than last)
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+                and duration between first and last positive is < 16 weeks.
+PERSADA -  Persistent ADA (either of):
+             INDUCED and Positive at the last post baseline result.
+             INDUCED and multiple post baseline positive (other than last)
```
```suggestion
             `INDUCED` and multiple post-baseline positive (other than last)
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+             INDUCED and multiple post baseline positive (other than last)
+                and duration between first and last positive is >= 16 weeks.
+TRANADAE  - Any Treatment Emergent Transient ADA  (either of): 
+             INDUCED and Positive at any Post Baseline other than Last
```
```suggestion
             `INDUCED` and Positive at any Post-Baseline other than last result
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+             INDUCED or ENHANCED and Positive at the last post baseline result.
+             INDUCED  or ENHANCED and multiple post baseline positive (other than last)
+ADABLPFL - Baseline ADA Evaluable - Baseline test exists (ABLFL = Y)
+ADPBFPFL - Post Baseline ADA Evaluable - At least one Valid (non-missing) post-baseline result.
```
```suggestion
`ADPBFPFL` - Post-Baseline ADA Evaluable - At least one Valid (non-missing) post-baseline result.
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+  select(-ADTM, -TFLAGV, -FANLDTM, -FANLDT, -LFLAGPOS, -FPPDT, -LPPDT)
+
+# Put PERSADA, TRANADA, INDUCED, ENHANCED, TDUR, ADADUR onto "is_flagdata" as "main_aab_pertran"
+#  Note: if get a signal_duplicate_records() error, then likely cause is  subject has a duplicate
```
```suggestion
# Note: signal_duplicate_records() error usually occurs when a subject has duplicate
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+# Put PERSADA, TRANADA, INDUCED, ENHANCED, TDUR, ADADUR onto "is_flagdata" as "main_aab_pertran"
+#  Note: if get a signal_duplicate_records() error, then likely cause is  subject has a duplicate
+#        record for a given BASETYPE, ADATYPE, ADAPARM and ISDTC. Investigate then add code
```
```suggestion
#        records for a given BASETYPE, ADATYPE, ADAPARM and ISDTC. Investigate then add code
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+## Assemble into Parameter Based Dataset {#parameters}
+
+In this section, we turn original IS records, the duplicated interpreted records plus
```
```suggestion
In this section, we turn original `IS` records, the duplicated interpreted records plus
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+## Assemble into Parameter Based Dataset {#parameters}
+
+In this section, we turn original IS records, the duplicated interpreted records plus
+each Analysis variable into separate prepared data frames.  These will then be
```
```suggestion
each Analysis variable into separate prepared data frames. These will then be
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+  select(-BASE, -CHG, -MRT, -DTL, -ADABLPFL, -ADPBLPFL)
+
+# By Visit Titer PBFLAGV data ---------------------
+# Note: By Visit parameters do not keep CHG, MRT and DTL
```
```suggestion
# Note: By Visit parameters do not keep CHG, MRT and DTL variables
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+
+In this section, we set together the desired parameter data sets created in the prior section.
+In this example, all are kept.  A specific study might not have Nab or need the by visit parameters (TFLAGV, PBFLAGV, ADASTATV)
+or need vars like ADADUR, LPPDTM and FPPDTM.
```
```suggestion
or need variables like `ADADUR`, `LPPDTM` and `FPPDTM`.
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+(ISBDAGNT is 'XANOMELINE' and ISBDAGNT is 'Y012345678').
+
+After merging the 'adab_param_data', final adjustments are made to PARAMCD and PARAM values.
+The adjustments are dependent on whether IS is based on SDTM v2.0 or prior to v1.8.
```
```suggestion
These adjustments depend on whether `IS` is based on SDTM v2.0 or an earlier version (prior to v1.8).
```

Fanny-Gautier commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+After merging the 'adab_param_data', final adjustments are made to PARAMCD and PARAM values.
+The adjustments are dependent on whether IS is based on SDTM v2.0 or prior to v1.8.
+
+Last step is computing ASEQ.  ASEQ should be computed last since the sort key includes
```
```suggestion
Last step is computing `ASEQ`. `ASEQ` should be computed last since the sort key includes
```

Kcdahn commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+The first part of the article describes the data staging, second part
+describes the methods of computation of parameters, the third part describes
+assembling he final data set into by visit and parametee based records 
+then adding final flags, ADSL variables then compute ASEQ.
```
OK.

Kcdahn commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1583 @@
+IN this first section, we compute ABLFL, BASE and CHG.
+VALIDBASE and VALIDPOST are assigned to accommodate if sponsor has restrictions
+to use directly or use later to qualify a record as baseline or post-baseline
+as analysis variables are computed.  For example, a missing value at baseline
```
Done.

bms63 commented:
> All. I have also been receiving comments in the main ADAB branch (2805_add_template_for_adab). There have been some code and comment updates there and maybe more coming. When the main ADAB is near final, I was thinking of then re-adding all the code one last time here then, rather then trying to keep them aligned manually (I am adding some however anyway).

yes agree - please finish template and let's merge in as final and then move onto the vignette.  

we wanted to label as experimental in 1.4 release and see if CDISC ADaM group would review once release and others in community.

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+umbrella of the [pharmaverse](https://github.com/pharmaverse). An
+example of applying metadata and perform associated checks can be found
+at the [pharmaverse E2E
+example](https://pharmaverse.github.io/examples/adam/adsl).
```
```suggestion
# Add Labels and Attributes {#attributes}

Note that attributes may not be preserved in some cases after processing
with `{admiral}`. The recommended approach is to apply variable labels
and other metadata as a final step in your data derivation process using
packages like:

-   [metacore](https://atorus-research.github.io/metacore/): establish a
    common foundation for the use of metadata within an R session.

-   [metatools](https://pharmaverse.github.io/metatools/): enable the
    use of metacore objects. Metatools can be used to build datasets or
    enhance columns in existing datasets as well as checking datasets
    against the metadata.

-   [xportr](https://atorus-research.github.io/xportr/): functionality
    to associate all metadata information to a local R data frame,
    perform data set level validation checks and convert into a
    [transport v5
    file(xpt)](https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.5/movefile/n1xbwdre0giahfn11c99yjkpi2yb.htm).

NOTE: Together with `{admiral}` these packages comprise an End to End
pipeline under the umbrella of the [pharmaverse](https://github.com/pharmaverse). An
example of applying metadata and perform associated checks can be found
at the [pharmaverse E2E
example](https://pharmaverse.github.io/examples/adam/adsl).

```

jeffreyad commented on _pkgdown.yml:
```diff
@@ -218,6 +218,8 @@ navbar:
         href: articles/questionnaires.html
       - text: Creating a PK NCA or Population PK ADaM
         href: articles/pk_adnca.html
+      - text: Creating an ADAB ADaM
```
```suggestion
      - text: Creating an Antidrug Antibody ADaM
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+This example presented uses underlying `EX` and `IS` domains where the
+`EX` and `IS` domains represent data as collected and the `ADAB` ADaM is
+output. For purposes of the examples, the surrogate data for ADA is called
+`IS_ADA`. `IS_ADA` is a product of the pharmaverse IS_ADA data. The `IS`
```
```suggestion
`IS_ADA`. `IS_ADA` is a product of the pharmaverse `IS_ADA` data. The `IS`
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+added to the `{pharmaversesdtm}` for ADA data.
+
+This template and the sample `IS` dataset works off of `IS` having primary
+Anti-Drug Antibody results (ex: `ISTESTCD` = 'ADA_BAB') and Neutralizing Antibody 
```
```suggestion
Anti-Drug Antibody results (ex: `ISTESTCD = "ADA_BAB"`) and Neutralizing Antibody 
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+
+This template and the sample `IS` dataset works off of `IS` having primary
+Anti-Drug Antibody results (ex: `ISTESTCD` = 'ADA_BAB') and Neutralizing Antibody 
+(ex:`ISTESTCD` = 'ADA_NAB') records present when 'ADA_BAB' has a positive finding.
```
```suggestion
(ex:`ISTESTCD = "ADA_NAB"`) records present when `ADA_BAB` has a positive finding.
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+(ex:`ISTESTCD` = 'ADA_NAB') records present when 'ADA_BAB' has a positive finding.
+
+One of the important aspects of the dataset is the separate derivation of
+original 'by visit' data, by visit interpreted results and parameter variables
```
```suggestion
original "by visit" data, by visit interpreted results and parameter variables
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+what was stored in pre-version 1.8 in `ISTESTCD` or `ISTEST` is now stored in `IS.ISBDAGNT`.
+This code provides a mechanism to process either version, see below comments for usage.
+
+If your `IS` dataset has values in `ISTESTCD` and `ISTEST` different then the example (or has 
```
```suggestion
If your `IS` dataset has different values in `ISTESTCD` and `ISTEST` than the example (or has 
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+If your `IS` dataset has values in `ISTESTCD` and `ISTEST` different then the example (or has 
+additional tests suitable for the `ADAB` domain), adjust this step accordingly. 
+
+`ADAPARM` and `ADATYPE` are assigned to assure consistent merge variables as values are calculated
```
```suggestion
`ADAPARM` and `ADATYPE` are assigned to ensure consistent merge variables as values are calculated
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+Common variables as such `AVISIT`, `AVISITN`, `FRLTU` are assigned.  
+
+`NFRLT` is nominal time from first dose. Common sources are `IS.VISITDY`. `IS.VISIT` information can
+also be for all values or to get unplanned visits (i.e. 'End of Study', 'Unscheduled').
```
```suggestion
also be for all values or to get unplanned visits (i.e. "End of Study", "Unscheduled").
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+
+## Prepare 'EX', Find First Dose and Derive AFRLT {#firstdose}
+
+At this step, we stage 'EX' data. Filter the data as desired to match 
```
```suggestion
At this step, we stage `EX` data. Filter the data as desired to match 
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+get the desired imputed time part. For example a common method is to use the time part from `IS.ISDTC` 
+(matched by the `IS` and `EX` date parts), then the `EX` missing time is adjusted for 
+pre-dose sampling planned time. User would add custom code for this.
+AENDTM from EX.EXENDTC is also computed in case needed.  Both dates then create
```
```suggestion
`AENDTM` from `EX.EXENDTC` is also computed in case needed.  Both dates then create
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+the standard interpreted result (`POSITIVE` or `NEGATIVE`) based values found
+in `IS.ISSTRESC` (these values will be sponsor specific as applicable, starter
+values supplied to demonstrate.). `RESULTN` is assigned `0`/`1` as the numeric version for `RESULTC`.
+RESULTN and RESULTC are pre-cursors to AVAL and AVALC and serve as interim stores
```
```suggestion
`RESULTN` and `RESULTC` are pre-cursors to `AVAL` and `AVALC` and serve as interim stores
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+BFLAG  - Baseline Positive/Negative Flag
+TFLAG  - Overall Treatment Emergent Category  (see *T-Flags)
+PBFLAG  - Post-Baseline Status
+ADASTAT - Overall Subject ADA Status (ADA Positive or Negative)
```
```suggestion
`TFLAGV` - By visit Treatment Emergent Category (see *T-Flags).
`PBFLAGV` - Post-Baseline Status by Visit
`ADASTATV` - Overall ADA Status by Visit (ADA Positive or Negative)
Subject Overall analysis variables
`BFLAG`  - Baseline Positive/Negative Flag
`TFLAG`  - Overall Treatment Emergent Category  (see *T-Flags)
`PBFLAG`  - Post-Baseline Status
`ADASTAT` - Overall Subject ADA Status (ADA Positive or Negative)
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+parent main analyte. Reminder that ADA and Nab are drug specific. Drug ABC can have ADA
+and optional Nab tests; Drug XYZ can also have the same pairs and are processed separately.
+
+The target variable for Nab is NABSTAT (subject Nab Status for the the matching parent analyte)
```
```suggestion
The target variable for Nab is `NABSTAT` (subject Nab Status for the the matching parent analyte)
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+The target variable for Nab is NABSTAT (subject Nab Status for the the matching parent analyte)
+In the example template, to be `NABSTAT = "Positive" or "Negative"`, the subject has to 
+`ADATAT="Positive"` on the main parent matching analyte. If not `ADASTAT  = "Positive"`, `NABSTAT`
+is set to either NA or 'MISSING" as desired.
```
```suggestion
is set to either NA or "MISSING" as desired.
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+This step is important as we want unique `PARAMCD`/`PARAM` values. The number suffixes start 
+with '1' and increment as the core parameter name is repeated. This could be due to having
+`RESULT` for ADA and Nab for Drug ABC. There could be two analytes/drug combos 
+(ISBDAGNT is 'XANOMELINE' and ISBDAGNT is 'Y012345678').
```
```suggestion
(`ISBDAGNT` is "XANOMELINE" and `ISBDAGNT` is "Y012345678").
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+`RESULT` for ADA and Nab for Drug ABC. There could be two analytes/drug combos 
+(ISBDAGNT is 'XANOMELINE' and ISBDAGNT is 'Y012345678').
+
+After merging the 'adab_param_data', final adjustments are made to PARAMCD and PARAM values.
```
```suggestion
After merging the `adab_param_data`, final adjustments are made to `PARAMCD` and `PARAM` values.
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1699 @@
+
+-   [Read in Source Data](#readdata)
+
+-   [Prepare 'EX', Find First Dose and Derive AFRLT](#firstdose)
```
```suggestion
-   [Prepare `EX`, Find First Dose and Derive `AFRLT`](#firstdose)
```

Kcdahn commented:
@bms63  I need the NFRLT  and xxTPT  PRs to get merged to main first.  I have an update pending commit so the vignette code and comments explains the new derive_var_nfrlt()  I added works.

bms63 commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1689 @@
+
+Note: It should be noted at this point that CDISC does not currently have
+a model or implementation guide for ADA (`ADAB`). This adab.R template
+is currently considered experimental.
```
```suggestion
Note:  CDISC does not currently have a model or implementation guide for ADA (`ADAB`). This `adab.R` template is currently considered **experimental**.
```

More concise

bms63 commented:
@Kcdahn functions are merged in - please proceed.

bms63 commented:
Looks like Code Styler actions is failing on the vignette

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1679 @@
+When `IS.ISSTRESC` is "POSITIVE", `IS.ISSTRESN` is commonly populated with values >0. 
+However, if `ISSTRESC` is "POSITIVE" with a missing `IS.ISSTRESN`, this template provides
+a method to assign `AVAL` using the assigned MRT value, sets `AVALC` to "N.C." (Not Calculable)
+and documents the operation by setting `DTYPE='MRT'`.  
```
```suggestion
and documents the operation by setting `DTYPE = "MRT"`.  
```

jeffreyad commented on vignettes/adab.Rmd:
```diff
@@ -0,0 +1,1679 @@
+adab_study <- bind_rows(adab_paramcds, adab_visits, adab_adadur, adab_fppdtm, adab_lppdtm)
+
+# Drop temporary variables and ADA Flag variables that are now parameterized, further below is a
+# final 'select' statement to to customize the final select.
```
```suggestion
# final `select` statement to to customize the final select.
```

Kcdahn commented:
I added the main adab.R template code updates related to derive_var_nfrlt()  updates and added/changed some related verbiage.
The Styler issue was some dataset_vignette() adds needed styling.
